OpenSim/Common/Delay.h

```diff
@@ -0,0 +1,179 @@
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ * Author(s): Chris Dembia                                                    *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "osimCommonDLL.h"
+#include "Component.h"
+
+namespace OpenSim {
+
+/** This operator delays its input by a given time delay. The value of the
+ * output at time t is the value the input had at time t-delay.
+ * A Delay can be used to model the delay in neural reflex circuits.
+ *
+ * For times prior to the start of a simulation this Component behaves as
+ * though the input value had been constant at its initial value.
+ *
+ * @tparam T The type of the quantity to be delayed. Common choices are a
+ * SimTK::Real (see ScalarDelay) or a SimTK::Vector (see VectorDelay).
+ *
+ * @see ScalarDelay, VectorDelay
+ *
+ *
+ * This Component has a single input named "input" and a single output named
+ * "output". The typical way to use this Component is as a subcomponent in
+ * another component. Here is a basic example in which a controller specifies
+ * the control signal for a single actuator using the delayed value of a
+ * coordinate:
+ * @code
+ * using namespace OpenSim;
+ * using namespace SimTK;
+ * class MyController : Controller {
+ * OpenSim_DECLARE_CONCRETE_OBJECT(MyController, Controller);
+ * public:
+ *     OpenSim_DECLARE_PROPERTY(delay, double, "Duration of delay (seconds).");
+ *     MyController() { constructProperties(); }
+ *     void computeControls(const State& s, Vector &controls) const override {
+ *         double q = _coordDelay.getValue(s);
+ *         // Delayed negative feedback control law.
+ *         double u = -10 * q;
+ *         getModel().getActuatorSet()[0].addInControls(Vector(1, u), controls);
```

Comment:
Not sure this is a good thing to have in an example -- I would hate to see people hard coding actuator indices in real code! For good pedagogy I think it would be worth a few more lines.